File: 69.x-的平方根.js
/*
 * @lc app=leetcode.cn id=69 lang=javascript
 *
 * [69] x 的平方根
 */

// @lc code=start
/**
 * @param {number} x
 * @return {number}
 */
var mySqrt = function (x) {
  if (isNaN(x)) return NaN;
  if (x === 0 || x === 1) return x;
  let val = x,
    last;
  do {
    last = val;
    val = (val + x / val) / 2;
  } while (Math.abs(val - last) >= Number.EPSILON);
  val = Math.floor(val);
  return val;
};

// @lc code=end

File: 69.x-的平方根.c
/*
 * @lc app=leetcode.cn id=69 lang=c
 *
 * [69] x 的平方根 
 */

// @lc code=start
#include <math.h>

int mySqrt(int x){
   double p=0.001;
   double val=1.0;double cheak;
    do
   {
        val = (x / val + val) / 2.0;
        cheak = val * val - x;
    } while((cheak >= 0 ? cheak : -cheak) > p);
    return floor(val);
}
// @lc code=end

